c10/core/ScalarType.h

```diff
@@ -317,7 +317,9 @@ static inline bool isSignedType(ScalarType t) {
       return std::numeric_limits<ctype>::is_signed;
 
     switch (t) {
-      AT_FORALL_SCALAR_TYPES_AND(Half, CASE_SIGNED)
+      AT_FORALL_SCALAR_TYPES_AND2(Half, Bool, CASE_SIGNED)
+      case ScalarType::BFloat16:
```

Comment:
Why is this one a special case?
Aren't there other special cases?

Reply:
Oh, I seemed to have missed quantized dtypes. Thank you for pointing this out.

Reply:
There is a `toUnderlying ` function just above that might help with handling quantized dtypes?

Reply:
I've fixed it by manually adding them, because I felt that using the macro would probably not reduce the size of the diff.

Reply:
Wouldn't adding `t = toUnderlying(t);` just before the switch work?

Reply:
Oh yes, why didn’t I think of that. Sorry, let me fix this.

Reply:
@albanD I've simplified it. Thanks for the review.